### ISSUE TITLE
Complete MP hangs by receiving unrelated message in active delete path [JIRA: RCS-271]

### ISSUE THREAD

**Issue by shino** [basho/riak_cs#1225]
Reproduction steps:
1. Set active delete threshold to 10KB or so
2. Call Put Object with 10KB object
3. Upload Multipart with 10MB object with 5MB part size
   (overly detailed. MP is sufficient here)

Then riak cs outputs log below and client hangs:

```
16:12:17.479 [error] Active deletion failed. Reason:
   {'$gen_sync_all_state_event',{<0.1710.0>,#Ref<0.0.0.112804>},stop}
```

---

Investigation so far:
1. In active delete path, there is raw `receive` with catch-all branch
2. In Complete MP path, manifest FSM is started and called by `new_mani_added` event'
   with {{UUID, AboutToCompletManifest}`.
3. If there is no parts to be deleted, the manifest FSM is ordered to be stoped.
4. But manifest fsm calls active delete logic and is doing some work,
   waiting message by raw `receive`.
5. `stop` event is received and error log is triggered
6. Caller of `stop` for the manifest fsm continues to wait.

**Comment by shino** [comment]
Fixed by #1227

**Comment by Basho-JIRA** [comment]
No release note needed as this is a bug during development cycle 

**_[posted via JIRA by Kota Uenishi]**_
